examples/step-57/step-57.cc

```diff
@@ -0,0 +1,878 @@
+    gamma(1.0),
+    degree(degree),
+    triangulation(Triangulation<dim>::maximum_smoothing),
+    fe(FE_Q<dim>(degree+1), dim,
+       FE_Q<dim>(degree),   1),
+    dof_handler(triangulation)
+  {}
+
+  // @sect4{StationaryNavierStokes::setup_dof}
+  // This function provide the information of DoF and initialize the constraints
+  // on current mesh.
+  template <int dim>
+  void StationaryNavierStokes<dim>::setup_dof()
+  {
+    system_matrix.clear();
+    pressure_mass_matrix.clear();
+
+    // The first step is to associate DoFs with a given mesh.
+    dof_handler.distribute_dofs (fe);
+
+    // We renumber the components to have all velocity DoFs come before the pressure
+    // DoFs to be able to use the block preconditioner.
+    //
+    std::vector<unsigned int> block_component(dim+1, 0);
+    block_component[dim] = 1;
+    DoFRenumbering::component_wise (dof_handler, block_component);
+
+    dofs_per_block.resize (2);
+    DoFTools::count_dofs_per_block (dof_handler, dofs_per_block, block_component);
+    unsigned int dof_u = dofs_per_block[0];
+    unsigned int dof_p = dofs_per_block[1];
+
+    // In Newton's scheme, we first apply the boundary condition on the solution
+    // obtained from the initial step. To make sure boundary condition satisfied,
+    // zero boundary condition is used for the Newton's update term. Therefore we
+    // set up two different constraint objects.
+    FEValuesExtractors::Vector velocities(0);
+    {
+      nonzero_constraints.clear();
+
+      DoFTools::make_hanging_node_constraints(dof_handler, nonzero_constraints);
+      VectorTools::interpolate_boundary_values(dof_handler,
+                                               0,
+                                               BoundaryValues<dim>(),
+                                               nonzero_constraints,
+                                               fe.component_mask(velocities));
+    }
+    nonzero_constraints.close();
+
+    {
+      zero_constraints.clear();
+
+      DoFTools::make_hanging_node_constraints(dof_handler, zero_constraints);
+      VectorTools::interpolate_boundary_values(dof_handler,
+                                               0,
+                                               ZeroFunction<dim>(dim+1),
+                                               zero_constraints,
+                                               fe.component_mask(velocities));
+    }
+    zero_constraints.close();
+
+
+
+    std::cout << "   Number of active cells: "
+              << triangulation.n_active_cells()
+              << std::endl
+              << "   Number of degrees of freedom: "
+              << dof_handler.n_dofs()
+              << " (" << dof_u << '+' << dof_p << ')'
+              << std::endl;
+  }
+
+  // @sect4{StationaryNavierStokes::initialize_system}
+  // On each mesh the structure of the linear system is different.
+  // This function initializes the matrix and vectors after mesh
+  // refinement.
```

Comment:
The STRUCTURE is actually the same. I think you want to say that the sparsity pattern and size of the linear system is different.